src/python/pants/engine/build_files.py

```diff
@@ -43,7 +43,7 @@ def _key_func(entry):
 
   The AddressFamily may be empty, but it will not be None.
   """
-  patterns = tuple(join(directory.path, p) for p in address_mapper.build_patterns)
+  patterns = (os.path.join(directory.path, p) for p in address_mapper.build_patterns)
```

Comment:
So, IIRC, not including `tuple` turns this into a generator, which means that it will be consumed on first-use. generators are (in general) error prone to put in variables like this, because it introduces sideeffects to consume them.

So, I would avoid doing this, unless you _want_ side-effecting behavior.

Reply:
Instead, I'll inline this to the `PathGlobs` constructor. `patterns` is never used again in this block. 

If we kept it as a variable, I agree this isn't worth the possible confusion.